lib/rootApi.js

```diff
@@ -387,3 +385,34 @@ api.getNodeIterator = callbackify(async (actor, options = {}) => {
 
   return iterator;
 });
+
+async function _findLedgerNode({actor, ledgerNodeId}) {
+  const query = {
+    id: database.hash(ledgerNodeId),
+    'meta.deleted': {
+      $exists: false
+    }
+  };
+  const record = await database.collections.ledgerNode.findOne(query, {});
+  if(!record) {
+    throw new BedrockError(
+      'Ledger node not found.',
+      'NotFound',
+      {httpStatusCode: 404, ledger: ledgerNodeId, public: true});
+  }
+  // only need to check permissions if the ledger is owned, otherwise it
+  // is public
+
+  // FIXME: is it safe to make this if(actor && record.ledgerNode.owner)
+  // what are important distinctions between undefined and null in this case?
+  if(record.ledgerNode.owner) {
```

Comment:
This should be safe:

```suggestion
  if(actor !== null && record.ledgerNode.owner) {
```

If an `actor` is undefined, the actor is anonymous, which is different from if `actor` is `null` -- in which case the permission check is skipped. Of course, this is a terrible distinction :), but it's legacy code... and once everything is moved to a capabilities based architecture we won't have to care about it anymore.